### PR TITLE
Fix improper usage in refactoring ns-policy

### PR DESCRIPTION
**Pull request by wuzhanpeng**
A simple fix for imporper usage during code refactoring. The cause of the issue can refer to the following PRs: #5371 #12078

No doc is required.

### CODE REVIEW

**Review by michaeljmarshall — APPROVED** [review]
LGTM. I see that the `isProducersExceeded` method in this class has the same logic that you're updating here. Should we fix that one too?

**Comment by michaeljmarshall** [comment]
@merlimat - PTAL

**Comment by wuzhanpeng** [comment]
> LGTM. I see that the `isProducersExceeded` method in this class has the same logic that you're updating here. Should we fix that one too?

@michaeljmarshall Yes, thank you for your reminder, but I prefer to open another issue later to discuss about `isProducersExceeded`. That involves some other issues.

**Comment by michaeljmarshall** [comment]
@wuzhanpeng - are you able to rebase your changes here? When I approved this previously, I wasn't a committer. I am now, so I can merge this if we get it into a good place.

**Review by eolivelli — APPROVED** [review]
great catch

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:214]: I think this one has been improved by hierarchy topic policy?

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

**Comment by tisonkun** [comment]
> When I approved this previously, I wasn't a committer. I am now, so I can merge this if we get it into a good place.

@michaeljmarshall I often rebase for the patch stale if I can nudge it forward. As a committer your can push to the PR branch as long as the author checks "allow edits from maintainers" (by default). Also, you may supersede the patch with your own while keep the original author in "Co-authored-by" footeer.

For this specific PR, https://github.com/apache/pulsar/pull/12158#discussion_r826926441 can be an effective request change :)

**Pull request closed without merging**
